mobilecoind/src/service.rs

```diff
@@ -4184,6 +4292,126 @@ mod test {
+                AccountKey::random(&mut rng).default_subaddress(),
+                swap_originator.default_subaddress(),
+                swap_counterparty.default_subaddress(),
+            ],
+            Amount::new(1_000_000_000_000, TokenId::from(1)),
+            &[KeyImage::from(101)],
+            &mut rng,
+        )
+        .unwrap();
+
+        // Insert into database.
+        let originator_monitor_id = mobilecoind_db.add_monitor(&swap_originator_data).unwrap();
+        let counterparty_monitor_id = mobilecoind_db.add_monitor(&swap_counterparty_data).unwrap();
+
+        // Allow the new monitor to process the ledger.
+        wait_for_monitors(&mobilecoind_db, &ledger_db, &logger);
+
+        // Get list of unspent tx outs
+        let utxos = mobilecoind_db
+            .get_utxos_for_subaddress(&originator_monitor_id, 0)
+            .unwrap();
+        assert!(!utxos.is_empty());
+
+        let offered_input = utxos
+            .iter()
+            .filter(|utxo| utxo.token_id == *Mob::ID)
+            .map(api::UnspentTxOut::from)
+            .next()
+            .unwrap();
+        let offered_value = offered_input.value;
+
+        // Generate a swap.
+        let generate_swap_response = {
+            let mut request = api::GenerateSwapRequest::new();
+            request.set_sender_monitor_id(originator_monitor_id.to_vec());
+            request.set_change_subaddress(0);
+            request.set_input(offered_input);
+            request.set_allow_partial_fill(true);
+            request.set_counter_value(123);
+            request.set_counter_token_id(1);
+            request.set_minimum_fill_value(10);
+            client.generate_swap(&request).unwrap()
+        };
+
+        // Now we will try to build a transaction that incorporates the swap.
+        // The counterparty needs to supply eUSD.
+        // Get list of unspent tx outs
+        let utxos = mobilecoind_db
+            .get_utxos_for_subaddress(&counterparty_monitor_id, 0)
+            .unwrap();
+        assert!(!utxos.is_empty());
+
+        // We will try to take half of the offered input.
+        let mut sci_for_tx = api::SciForTx::new();
+        sci_for_tx.set_sci(generate_swap_response.get_sci().clone());
+        sci_for_tx.set_partial_fill_value(offered_value / 2);
+
+        // Try to add the swap to a mixed transaction
+        let generate_mixed_tx_response = {
+            let mut request = api::GenerateMixedTxRequest::new();
+            request.set_sender_monitor_id(counterparty_monitor_id.to_vec());
+            request.set_change_subaddress(0);
+            request.set_input_list(utxos.iter().map(Into::into).collect());
+            request.set_scis(vec![sci_for_tx].into());
+            client.generate_mixed_tx(&request).unwrap()
+        };
+
+        assert_eq!(
+            generate_mixed_tx_response
+                .get_tx_proposal()
+                .get_scis()
+                .len(),
+            1
+        );
```

Comment:
i would like to try to unmask the outputs of the Tx and see that they make sense, going to extend this test